src/NTestController/NTestController.cs

```diff
@@ -56,16 +58,37 @@ public static void Main(string[] args)
                     // Load plugins.
                     Dictionary<PluginType, IPlugin> plugins = GetPlugins(_options.ConfigFile);
 
+                    var xmlDoc = XmlUtils.LoadXmlDocument(_options.ConfigFile);
+                    var defaultsNode = GetDefaultsXmlNode(xmlDoc);
+                    var platforms = GetPlatforms(xmlDoc, defaultsNode);
+
                     // Execute Reader plugin.
-                    plugins[PluginType.TestReader].Execute();
+                    var readerPlugin = plugins[PluginType.TestReader] as IReaderPlugin;
+                    readerPlugin.TestInputFile = _options.TestFile;
+                    readerPlugin.Execute();
+
+                    // Add all tests to queue.
+                    var testQueue = new TestQueue();
+
+                    foreach (var test in readerPlugin.Tests)
+                    {
+                        testQueue.EnqueueTestToRun(test);
+                    }
 
                     // Execute Setup plugin.
 
                     // Execute Test Executor plugin.
+                    var executorPlugin = plugins[PluginType.TestExecutor] as IExecutorPlugin;
+                    executorPlugin.Options = _options;
+                    executorPlugin.TestQueue = testQueue;

```

Comment:
Another option other than passing these properties to the plugins in here would be to pass 'this' NTestController to the plugins in their factories, then the plugins can copy whatever properties they need from the NTestController.